lib/galaxy/model/store/__init__.py

```diff
@@ -2200,6 +2227,224 @@ def _init_crate(self):
         return ro_crate
 
 
+class WorkflowInvocationOnlyExportStore(DirectoryModelExportStore):
+    def export_history(self, history: model.History, include_hidden: bool = False, include_deleted: bool = False):
+        """Export history to store."""
+        raise NotImplementedError()
+
+    def export_library(self, history, include_hidden=False, include_deleted=False):
+        """Export library to store."""
+        raise NotImplementedError()
+
+    @property
+    def only_invocation(self) -> model.WorkflowInvocation:
+        assert len(self.included_invocations) == 1
+        return self.included_invocations[0]
+
+
+@dataclass
+class BcoExportOptions:
+    galaxy_url: str
+    galaxy_version: str
+    merge_history_metadata: bool = False
+    override_environment_variables: Optional[Dict[str, str]] = None
+    override_empirical_error: Optional[Dict[str, str]] = None
+    override_algorithmic_error: Optional[Dict[str, str]] = None
+    override_xref: Optional[List[XrefItem]] = None
+
+
+class BcoModelExportStore(WorkflowInvocationOnlyExportStore):
+    def __init__(self, uri, export_options: BcoExportOptions, **kwds):
+        temp_output_dir = tempfile.mkdtemp()
```

Comment:
where is this tempdir created? I guess it will not be large and does not matter much, but should it be in the some Galaxy temp dir?

Reply:
It will be part of new_file_path, we're patching the basedir when setting up `GalaxyAppConfiguration`